### ISSUE TITLE
Amplify Auth Signout with Multiple Tabs

### ISSUE THREAD

**Issue by rkushkuley** [aws-amplify/amplify-js#10300]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

Authentication

### Amplify Categories

auth

### Environment information

<details>

```
"aws-amplify": "^4.3.24",
```

</details>

### Describe the bug

When you have 2 tabs open for the same authenticated user and you try to sign out from one tab it gets stuck (without an error) until the other tabs calls Auth.signOut().

Once the other tab calls Auth.signOut() it successfuly signs out from both tabs.
![image](https://user-images.githubusercontent.com/33210497/188986754-50e75240-49ce-41b6-b341-7a83a11b8bf9.png)

### Expected behavior

I expect both or one tab to signout or at least an error to be logged. Right now it is just waiting.  

### Reproduction steps

1. Sign in
2. copy and past link to new tab (you now have 2 tabs signed in (they are sharing localstorage so makes sense)
3. call Auth.SignOut() -> does not sign out until you call Auth.SignOut() on the other tab 

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by tannerabread** [comment]
@rkushkuley were you using the Amplify UI components for this? When I tried to recreate following your steps it signed out for me on the original page and a refresh of the other page signed that one out too

**Comment by rkushkuley** [comment]
No UI components, just calling Auth.signOut() on a button click

**Comment by tannerabread** [comment]
We need a little more information to try to reproduce this issue

Is this still occurring for you?
If so, what browser are you using? 
Are you using social-signIn or just the standard username/password signIn with Auth.signIn()?

**Comment by rkushkuley** [comment]
Yes this is still occuring. Using chrome and Auth.signIn(). I actually did notice that the refresh signs out both tabs. But the first tab that actually called the Auth.signOut doesnt sign out until tab 2 is refreshed.

**Comment by rkushkuley** [comment]
Another way to show the issue is that if you have the 2 tabs. Auth.signOut on tab 1 doesnt clear the local storage with the cognito ids until tab 2 is refreshed or Auth.signOut is called on tab 2.

**Comment by rkushkuley** [comment]
Maybe its worth using sessionStorage rather than localStorage?

**Comment by rkushkuley** [comment]
update: the issue was that i was calling DataStore.stop and DataStore,clear before Auth.SignOut(). i do want to clear the datastore on signout is there a better way?

`await DataStore.stop();
await DataStore.clear();
await Auth.signOut();
history.push('/signin')`

**Comment by tannerabread** [comment]
Glad you were able to find the issue! I was having trouble reproducing 

Check out the code example mentioned here: https://docs.amplify.aws/lib/datastore/sync/q/platform/js/#clear-local-data

This should clear your local DataStore as you are signing out

**Comment by rkushkuley** [comment]
this works great thank you!!! is there anyway to share events between tabs using amplify? so if one tab is logged out the other tab gets logged out as well?

**Comment by tannerabread** [comment]
> is there anyway to share events between tabs using amplify? so if one tab is logged out the other tab gets logged out as well?

There is no built-in way with amplify. One amplify instance emits events to the page it was initialized on. 

However, tabs do share localStorage so an onFocus listener could check for whether the user is still signed in or not. A couple of ways this can be done is either by looking for tokens in localStorage or attempting to call `Auth.currentAuthenticatedUser()` when switching between tabs and redirect the user to sign in.

Vercel does something similar with [SWR](https://swr.vercel.app/docs/revalidation) if you want to check that out
